### DISCUSSION TITLE
How to access environment variables when using Webpack 5?

### DISCUSSION THREAD

**Discussion by jensmeindertsma** [vercel/next.js#15688] [Help] [answered]
In Webpack 5 `process` is no longer defined. @timneutkens recently added a polyfill for this, but he stated "_this is to not break existing projects_".

My question is: how are new projects supposed to use environment variables in Webpack 5? The docs still recommend using `process.env.*`, but they might not have been updated yet. 

I understand I can use the polyfill, but surely Webpack 5 doesn't just remove environment variables altogether? Under what accessor are they now available?

**Comment by Timer** [comment, marked as answer]
Please upgrade to Next.js 9.5.1!

**Reply by jensmeindertsma** [reply]
I've done so and this does solve problems for existing projects. I just wanted to ask whether the removal of `process` from Webpack 5 means I should be accessing my environment variables in a different way that I'm not aware of. If using `process` is still recommended ( also for new projects ), that's fine. I'm very grateful for all these updates 🙏

**Reply by jensmeindertsma** [reply]
I want to clarify: I felt like the removal of `process` from Webpack 5 meant that it was now "deprecated".
